### PR TITLE
User's repository

### PR DESCRIPTION
**Pull request by mpeck**
Moves user bits into a users repository

prereq for #671

### CODE REVIEW

File: lib/cog/repository/users.ex [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  @doc """
+  Deletes a user
+  """
+  @spec delete(%User{}) :: {:ok, %User{}} | {:error, :not_found}
+  def delete(%User{}=user) do
+    try do
+      Repo.delete(user)
+    rescue
+      Ecto.StateModelError ->
```
Comment (vanstee) [inline lib/cog/repository/users.ex]: Is this supposed to be `StaleModelError`? Also surprised that the `delete` func raises like this. Did you mean `delete!`?

Comment (mpeck) [inline lib/cog/repository/users.ex]: Actually we may not need this. I copied it from the triggers repository. I think it only get's triggered if you try to delete the same model twice.

Comment (mpeck) [inline lib/cog/repository/users.ex]: I'll just drop it for now.

**Comment by christophermaier** [comment]
👍

**Pull request merged**